src/main/java/com/mongodb/kafka/connect/source/schema/BsonDocumentToSchema.java

```diff
@@ -163,6 +167,13 @@ private static Schema combinedSchema(final Schema firstSchema, final Schema seco
                     structBuilder.field(
                         name,
                         combineFieldSchema(firstSchema.field(name), secondSchema.field(name))));
+
+        logIncompatibleSchemasIf(
+            () ->
+                structBuilder.fields().stream()
```

Comment:
This seems like an expensive check to do when logging is enabled, in the case where the it evaluates to false.  But I can't think of a way to avoid it and still get the effect you're looking for.

I'm also wondering how much logging could happen for the same struct.  Say we have an array with 20 documents who's schema need to be combined.  Will this log once for every attempt to combine (i.e. 19 times)?